test/common/network/listen_socket_impl_test.cc

```diff
@@ -94,11 +96,6 @@ class ListenSocketImplTest : public testing::TestWithParam<Address::IpVersion> {
       // The address and port are bound already, should throw exception.
       EXPECT_THROW(createListenSocketPtr(addr, options2, true), SocketBindException);
 
-      // Test the case of a socket with fd and given address and port.
-      IoHandlePtr dup_handle = std::make_unique<IoSocketHandleImpl>(dup(socket1->ioHandle().fd()));
-      auto socket3 = createListenSocketPtr(std::move(dup_handle), addr, nullptr);
-      EXPECT_EQ(addr->asString(), socket3->localAddress()->asString());
-
```

Comment:
Should this just be disabled for Win32?

Reply:
No. That test logic is a complete no-op, our test goal is not to confirm that linux ::dup() works. The result is never used.

Reply:
(We had the same initial reaction, but as we looked at that test pattern, there appears to be no purpose and no reason to implement os_sys_calls.dup() at this time.)

Reply:
Isn't this testing the `Envoy::Network::NetworkListenSocket` constructor that takes an `IoHandlePtr` vs. `Address::InstanceConstSharedPtr`. There is probably another way to test that without using `::dup()`, though.

Reply:
we addressed this by removing dup and closing the fd before creating a new socket and using that to construct an `IoHandlePtr`

Reply:
resolving as we addressed the removal of `dup`